client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -541,10 +541,38 @@ class CalypsoifyIframe extends Component< Props & ConnectedProps & ProtectedForm
 			: `Block Editor > ${ postTypeText } > New`;
 	};
 
-	onIframeLoaded = ( iframeUrl: string ) => {
+	onIframeLoaded = async ( iframeUrl: string ) => {
 		if ( ! this.successfulIframeLoad ) {
-			window.location.replace( iframeUrl );
-			return;
+			// Sometimes (like in IE) the WindowActions.Loaded message arrives
+			// after the onLoad event is fired. To deal with this case we'll
+			// poll `this.successfulIframeLoad` for a while before redirecting.
+			let cancelPolling = false;
+
+			// Checks to see if the iFrame has loaded every 200ms. If it has
+			// loaded, then resolve the promise.
+			const pollForLoadedFlag = new Promise( resolve => {
+				const waitForSuccessfulLoad = () => {
+					if ( this.successfulIframeLoad ) {
+						resolve();
+						return;
+					}
+
+					! cancelPolling && setTimeout( waitForSuccessfulLoad, 200 );
```

Comment:
I find it interesting we are recursively calling `setTimeout` here as opposed to using an Interval and clearing it.  It looks good and works well, just curious if there was any specific reason to avoid an interval here?

Reply:
good point! I changed it to interval instead of timeout, which makes it a bit more readable :)

Reply:
fwiw I reach for `setTimeout` as a first instinct because:
- I find it easier to cancel the loop
- If the callback function is async, then theoretically the callback could take longer to run than the the interval duration

Those aren't issues in this case though :)